enumerable_methods.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |a|
+      arr.push(a) if yield(a)
+    end
+    arr
+  end
+
+  # my_all
+  def my_all?(arg = nil) # rubocop:disable Layout/TrailingWhitespace
+    if block_given?
+      my_each { |a| return false unless yield(a) }
+      return true
+    end
+    if arg.class == Regexp
+      my_each { |a| return false unless a.match(arg) }
+      return true
+    elsif arg.class == Class
+      my_each { |a| return false unless a.class == arg || a.class.superclass == arg }
+      return true
+    elsif arg.nil? && !block_given? && !empty?
+      return false
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |a| return true if yield(a) }
+      return false
+    end
+    if arg.class == Regexp
+      my_each { |a| return true if a.match(arg) }
+      return false
+    elsif arg.class == Class
+      my_each { |a| return true if a.class == arg || a.class.superclass == arg }
+      return false
+    elsif arg.nil? && !block_given? && !empty?
+      return true
+    end
+    false
```

Comment:
**[REQUIRED]**
* when no block or argument is given returns true if at least one of the collection is not false or nil.
![any1](https://user-images.githubusercontent.com/44072711/68568471-a6fd2680-0429-11ea-86d8-c854aa7b5e38.png)
* when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
       expected true
            got false

```